File: WoodPuzzle/src/woodPuzzle/model/Coordinate.java
package woodPuzzle.model;

public class Coordinate {
	public final int x; // width
	public final int y; // height
	public final int z; // length
	
	public Coordinate(int x, int y, int z) {
		this.x = x;
		this.y = y;
		this.z = z;
	}
	
	public static Coordinate buildCoordinate(String text) {
		String[] result = text.split(",");
		if (result.length < 3) 
			return null;
		return new Coordinate(Integer.parseInt(result[0]), Integer.parseInt(result[1]), Integer.parseInt(result[2]));
	}		
}

File: src/woodpuzzle/model/Coordinate.kt
package woodpuzzle.model

/**
 * Creates a coordinate with the given x, y, and z components.
 * @param x The x component.
 * @param y The y component.
 * @param z The z component.
 */
data class Coordinate (
    val x: Int, // width
    val y: Int, // height
    val z: Int  // length
) {
    /**
     * Creates a coordinate that has the usual vector
     * addition applied to it.
     * @param x The x component.
     * @param y The y component.
     * @param z The z component.
     * @return The transformed coordinate (a new coordinate instance).
     */
    fun vectorAdd(x: Int, y: Int, z: Int): Coordinate =
        Coordinate(this.x + x, this.y + y, this.z + z)
}